exporter/api.py

```diff
@@ -55,19 +106,24 @@ def export_resources_to_tarball(learning_resources, username):
             LearningResources to export in tarball
         username (unicode): Name of user
     Returns:
-        unicode: path of tarball within django-storage
+        (unicode, bool):
+            First item is path of tarball within django-storage.
+            Second item is whether a static asset collision was detected.
     """
     handle, tempfilepath = mkstemp()
     os.close(handle)
     try:
-        archive = tarfile.open(tempfilepath, mode='w:gz')
-
-        tempdir = export_resources_to_directory(learning_resources)
+        oldcwd = os.getcwd()
+        tempdir, collision = export_resources_to_directory(learning_resources)
         try:
-            for name in os.listdir(tempdir):
-                abs_path = os.path.join(tempdir, name)
-                if os.path.isfile(abs_path):
-                    archive.add(abs_path, arcname=name)
+            os.chdir(tempdir)
+            archive = tarfile.open(tempfilepath, mode='w:gz')
+
+            for root, _, files in os.walk("."):
+                for name in files:
+                    abs_path = os.path.join(root, name)
+                    abs_path = re.sub(r'^\./', '', abs_path)
```

Comment:
ACK. I'm not one of those people who has a problem with a liberal use of regular expressions. I just thought it might be a tad clearer.

Maybe compile the regex at the top of the function with a comment that it strips off the leading `./`, then use the compiled regex in the loop.